lib/schema_synchroniser.rb

```diff
@@ -0,0 +1,15 @@
+class SchemaSynchroniser
+  attr_reader :errors
+
+  def initialize(index_group)
+    @index = index_group.current
+  end
+
+  def sync
```

Comment:
Another general comment: Given this class looks a lot like a command/interactor type thing, I'd name the "main" method of it `#call` which is more conventional these days in the Ruby world (as the class name already says what it does anyway).

This can give you lots of nifty advantages around DI and testing ([this is a good article that mostly aligns with my thoughts](https://www.alchemists.io/articles/command_pattern)) and makes it clearer that this is a single-responsibility command object.

Reply:
Done. Thanks for the link, interesting read